### ISSUE TITLE
Featured images not loading when viewing posts on a private site

### ISSUE THREAD

**Issue by guarani** [wordpress-mobile/WordPress-iOS#16607]
### Expected behavior

When viewing a post (e.g. My Site → Posts → tap View on a post) that has a featured image set, that image should load correctly in the app's web view. This should work even if the site is private and I tested that this does work fine on WordPress for Android.

### Actual behavior

On private sites, when viewing a post that has a featured image set, that image does **not** load correctly in the app's web view.

### Steps to reproduce the behavior

1. Log in to an account and locate a private site (My Site → Site Settings → Privacy is set to **Private**)
2. View a post with a featured image set: My Site → Posts → tap View on the post
3. When the web view opens, notice that the featured image does not load
4. If you inspect the app's web view in Safari (which may be only possible when running the app directly from Xcode), the console reveals some errors:

```
    Failed to load resource: the server responded with a status of 403 ()
    Unable to post message to https://public-api.wordpress.com. Recipient has origin https://widgets.wp.com.
 ```

<img width="350" alt="Screen Shot 2021-06-01 at 16 01 31" src="https://user-images.githubusercontent.com/1898325/120383231-a3264300-c2f2-11eb-856b-3a5b25fb8691.png">

Note images are cached so if you can't reproduce this, try a clean install.

**Comment by hypest** [comment]
Might be related to https://github.com/wordpress-mobile/WordPress-iOS/issues/15596 too? @startuptester , I've added this ticket to the Groundskeeping queue but, not sure at which position I should move it to (the ToDo column currently has 211 tickets). Can you help with figuring out the prioritization compared to the other tickets on the ToDo column? Thanks!

**Comment by guarani** [comment]
> Might be related to #15596 too?

Good catch! I think it is a duplicate because I just confirmed that this issue only seem to happens on private sites with **_custom domains_**, which is also noted on that issue.

I added [a note](https://github.com/wordpress-mobile/WordPress-iOS/issues/15596#issuecomment-853036290) to #15596 to include this `Blog Posts → View` web view in its scope.

I'm happy to mark this as closed, but I'll just wait for confirmation from you, @startuptester, first.
